src/views/lighttable.c

```diff
@@ -221,6 +233,15 @@ static void switch_layout_to(dt_library_t *lib, int new_layout)
     lib->offset_x = 0;
     lib->offset_y = 0;
   }
+
+  dt_lib_module_t *m = darktable.view_manager->proxy.filmstrip.module;
+  gtk_widget_show(GTK_WIDGET(m->widget));
+
+  if(new_layout == DT_LAYOUT_EXPOSE) {
```

Comment:
You want to do that in expose() and not here. The switch_layout_to() is called only when changing the mode but the first time we open dt we do want this to happen so in expose() we are sure that the proper action will be taken each time.

Reply:
switch_layout()_to is called in expose() if(lib->layout != new_layout). lib->layout is initialised to -1. So it is called the first time in expose.